TERMINOLOGY.md

```diff
@@ -0,0 +1,53 @@
+# Terminology
+
+## Estimators and Predictors
+Fairlearn largely follows the [terminology established by scikit-learn](https://scikit-learn.org/stable/developers/contributing.html#different-objects), specifically:
+- Estimator: implements a `fit` method
+- Predictor: implements a `predict` method
+However, contrary to [scikit-learn estimators](https://scikit-learn.org/stable/glossary.html#term-estimator) we do not require determinism. Lots of unfairness mitigation techniques require randomized output to meet their objective in expectation. We do provide unrandomized versions of every method as well, but they may not be able to uphold the same theoretical guarantees.
```

Comment:
Rather than "do not require determinism" would it be better to say that "in fairlearn, estimators may not be deterministic." Also rather than "contrary to scikit-learn" it might be better to say "Unlike scikit-learn" since we aren't in opposition to sklearn. The "However" isn't necessary.